GAME_DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,122 @@
+- Travelling
+- Parkour
+- Mining
+- Building
+- Crafting
+- Managing inventory
+- Fighting
+
+These moments are extremely important as they let the player mentally rest, so making sure they're as satisfying and consistent as possible is a must.
+
+### Make the World Feel Alive
+To add immersion to the game, creatures should perform behaviors outside of player input, such as hunting, migrating, eating, or sleeping.
+
+# Balancing
+When balancing the game, keep in mind how players might interact with the world, the wildlife, and each other.
+
+### 2OP4ME
+At no point should the player be extremely hard to kill. Armor, tools, accessories, and buffs should be designed around aiding the player in skill-based encounters, not making them win regardless of skill.
+
+### Trade-offs
+If the player is given something to aid them, then it should have an appropriate take-away to balance it.
+
+**Examples:**
+- Using rare resources to create a strong tool.
+- An accessory that heals the player, but takes away energy or some other resource.
+- Enemies have strengths and weaknesses towards particular damage types
+
+### Mob Greifing
+Creatures should not be able to grief players without their input. Destructive creatures are reserved for caves.
+
+# Little Details
+The following contains spoilers for future content. If you are not looking into contributing, you should not read this section.
+<details>
+  <summary>Open Spoilers</summary>
+
+### Big Trees vs Small Trees
+There are two categories of trees, big and small. Big trees are designed to be built upon or left as decoration, whereas small trees are designed to be chopped down.
+
+### World Building with Gameplay
+Gameplay elements have implications on the world's lore, and the same goes other way around. It's best to design the gameplay element first, and then apply the world's style to the element during implementation.
+
+### Vegetation
+Vegetation should always fit the biome's climate. For example, Toadstools prefer humid areas, while Boletes prefer nutritious areas. Think about how a plant would fit into a biome or structure.
+
+### Caves are Creepy and Mysterious
+As the player goes deeper into Cubyz, they'll find that the music gets scarier, the monsters get harder and more disturbing, and the cave generation becomes an utter spectacle. We want the player to feel uneasy and stressed as they go down because it makes finding underground resources feel more rewarding.
+
+### The Root
+The Root is a cave located at a depth of -50000, sprawling with stone roots.
+
+This area is not required for progression, instead acting as a bonus challenge area that grants the player appropriate rewards for their current place in progression. It is the only area in the game (as of now) that will require a progression step to access, being an accessory that completely nullifies fall damage. As the player goes deeper, they'll find more curious areas that will put various skills to the test.
```

Comment:
> This area is not required for progression

In my mind the root is the last step of the progression system. It is meant to be reachable in a normal play-through and I think we can make a couple of fun and mechanically unique bosses down there as well.

Reply:
Yes, that should be an issue, and yeah I would love to have more fun post-endgame areas.
Though I suppose we should probably focus on the areas that the player can actually reach first.